x/poe/types/genesis.go

```diff
@@ -16,240 +16,242 @@ const DefaultBondDenom = "utgd"
+					Quorum:        sdk.NewDec(51),
+					Threshold:     sdk.NewDec(55),
+					AllowEndEarly: true,
+				},
+			},
+			CommunityPoolContractConfig: &CommunityPoolContractConfig{
+				VotingRules: VotingRules{
+					VotingPeriod:  21,
+					Quorum:        sdk.NewDec(10),
+					Threshold:     sdk.NewDec(60),
+					AllowEndEarly: true,
+				},
+			},
+			ValidatorVotingContractConfig: &ValidatorVotingContractConfig{
+				VotingRules: VotingRules{
+					VotingPeriod:  14,
+					Quorum:        sdk.NewDec(40),
+					Threshold:     sdk.NewDec(66),
+					AllowEndEarly: true,
+				},
 			},
-			DisputeCost: sdk.NewCoin(DefaultBondDenom, sdk.NewInt(1_000_000)),
+			ArbiterPoolContractConfig: &ArbiterPoolContractConfig{
+				Name:         "Arbiter Pool",
+				EscrowAmount: sdk.NewCoin(DefaultBondDenom, sdk.NewInt(1_000_000)),
+				VotingRules: VotingRules{
+					VotingPeriod:  30,
+					Quorum:        sdk.NewDec(51),
+					Threshold:     sdk.NewDec(55),
+					AllowEndEarly: true,
+				},
+				DisputeCost: sdk.NewCoin(DefaultBondDenom, sdk.NewInt(1_000_000)),
+			},
+			SystemAdminAddress: sdk.AccAddress(rand.Bytes(address.Len)).String(),
 		},
-		SystemAdminAddress: sdk.AccAddress(rand.Bytes(address.Len)).String(),
-		Params:             DefaultParams(),
 	}
 }
 
+// ValidateGenesis validates genesis for PoE module
 func ValidateGenesis(g GenesisState, txJSONDecoder sdk.TxDecoder) error {
-	if g.SeedContracts {
+	if err := g.Params.Validate(); err != nil {
+		return sdkerrors.Wrap(err, "params")
+	}
+	if g.SeedContracts != nil {
+		// seed genesis
 		if len(g.Contracts) != 0 {
 			return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "seed enabled but PoE contracts addresses provided")
 		}
-		if len(g.Engagement) == 0 {
-			return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "empty engagement group")
+		if err := validateSeedContracts(g.SeedContracts, txJSONDecoder); err != nil {
+			return sdkerrors.Wrap(err, "seed contracts")
 		}
-		if g.EngagementContractConfig == nil {
-			return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "empty engagement contract config")
-		}
-		if err := g.EngagementContractConfig.ValidateBasic(); err != nil {
-			return sdkerrors.Wrap(err, "engagement contract config")
+		return nil
+	}
+	// dump import genesis
+	if len(g.Contracts) == 0 {
+		return sdkerrors.Wrap(wasmtypes.ErrInvalidGenesis, "no PoE contract addresses provided")
+	}
+	uniqueContractTypes := make(map[PoEContractType]struct{}, len(g.Contracts))
+	for i, v := range g.Contracts {
+		if err := v.ValidateBasic(); err != nil {
+			return sdkerrors.Wrapf(err, "contract %d", i)
 		}
-		if err := sdk.ValidateDenom(g.BondDenom); err != nil {
-			return sdkerrors.Wrap(err, "bond denom")
+		if _, exists := uniqueContractTypes[v.ContractType]; exists {
+			return sdkerrors.Wrapf(wasmtypes.ErrDuplicate, "contract type %s", v.ContractType.String())
 		}
+		uniqueContractTypes[v.ContractType] = struct{}{}
+	}
+	if len(uniqueContractTypes) != len(PoEContractType_name)-1 {
```

Comment:
Very good check after loading.

Why the -1 btw? We shouldn't be doing any auto-creation on init in this state import phase

Reply:
it is -1 because one element is used for the `UNDEFINED` value, which is not used.